src/compiler/transformers/declarations/emitBinder.ts

```diff
@@ -0,0 +1,913 @@
+ */
+export function getMemberKey(name: string | Exclude<PropertyName, ComputedPropertyName> | NoSubstitutionTemplateLiteral): MemberKey;
+/**
+ * Gets the symbolic name for a member from its type.
+ * @internal
+ */
+export function getMemberKey(name: string | PropertyName | NoSubstitutionTemplateLiteral | undefined): MemberKey | undefined;
+export function getMemberKey(name: string | PropertyName | NoSubstitutionTemplateLiteral | undefined): string | undefined {
+    if (name === undefined) {
+        return undefined;
+    }
+    if (typeof name === "string") {
+        return ("I:" + name);
+    }
+    if (isPrivateIdentifier(name)) {
+        return ("P:" + name.escapedText);
+    }
+    if (isIdentifier(name)) {
+        return ("I:" + name.escapedText);
+    }
+    if (isStringLiteralLike(name)) {
+        return ("I:" + name.text);
+    }
+    if (isNumericLiteral(name)) {
+        return ("I:" + (+name.text));
+    }
+    if (isComputedPropertyName(name)) {
+        let computedName = name.expression;
+
+        if (isStringLiteralLike(computedName)) {
+            return ("I:" + computedName.text);
+        }
+        if (isNumericLiteral(computedName)) {
+            return ("I:" + (+computedName.text));
+        }
+        if (
+            isPrefixUnaryExpression(computedName)
+            && isNumericLiteral(computedName.operand)
+        ) {
+            if (computedName.operator === SyntaxKind.MinusToken) {
+                return ("I:" + (-computedName.operand.text));
+            }
+            else if (computedName.operator === SyntaxKind.PlusToken) {
+                return ("I:" + (+computedName.operand.text));
+            }
+            else {
+                return undefined;
+            }
+        }
+        let fullId = "C:";
+        // We only support dotted identifiers as property keys
+        while (true) {
+            if (isIdentifier(computedName)) {
+                fullId += computedName.escapedText;
+                break;
+            }
+            else if (isPropertyAccessExpression(computedName)) {
+                fullId += computedName.name.escapedText;
+                computedName = computedName.expression;
+            }
+            else {
+                // Can't compute a property key, bail
+                return undefined;
+            }
+        }
+        return fullId;
+    }
+    return undefined;
+}
+
+/**
+ * Gets the symbolic name for a member from its type.
+ * @internal
+ */
+export function getMemberKeyFromElement(element: NamedDeclaration): MemberKey | undefined {
+    if (isConstructorDeclaration(element) || isConstructSignatureDeclaration(element)) {
+        return "@constructor" as MemberKey;
+    }
+    const name = element.name;
+    if (!name || isElementAccessExpression(name) || isObjectBindingPattern(name) || isArrayBindingPattern(name) || isJsxNamespacedName(name) || isPropertyAccessExpression(name)) return undefined;
+    return getMemberKey(name) as MemberKey;

```

Comment:
From my skimming, `getMemberKey` already returns `MemberKey`; is there a reason for the cast here?

Reply:
Removed.